src/plugins/cuda_gds/gds_backend.cpp

```
@@ -102,19 +104,41 @@ nixl_status_t nixlGdsEngine::registerMem(const nixlBlobDesc &mem,
+                md->owned = true;
+                md->path = spec->path;
+                // NB: gds_file_map keys on fd, so two path-mode regs of the
+                // same path get two cuFile handles (no dedup). Acceptable
+                // for python clients that register one file per descriptor;
+                // callers that need dedup should open the fd themselves and
+                // use fd-in-devId registration.
+            }
+
+            // Check if we already have a file handle for this fd
+            auto it = gds_file_map.find(fd);
             if (it != gds_file_map.end()) {
```

Cppcheck (2.20.0)

[style] The function 'getSupportedMems' is never used.

(unusedFunction)